java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java

```diff
@@ -1,23 +1,147 @@
 // Copyright (c) Microsoft. All rights reserved.
 package com.microsoft.semantickernel.orchestration.contextvariables;
 
+import com.microsoft.semantickernel.exceptions.SKException;
+import com.microsoft.semantickernel.orchestration.contextvariables.ContextVariableTypeConverter.NoopConverter;
+import javax.annotation.Nullable;
+
 public class ContextVariable<T> {
 
     private final ContextVariableType<T> type;
+
+    @Nullable
     private final T value;
 
-    public ContextVariable(ContextVariableType<T> type, T value) {
+    public ContextVariable(ContextVariableType<T> type,
+        @Nullable T value) {
         this.type = type;
         this.value = value;
     }
 
+    public static <T> ContextVariable<T> of(Class<T> type, @Nullable T t) {
+        if (t == null) {
+            return new ContextVariable<>(
+                ContextVariableTypes.getDefaultVariableTypeForClass(type),
+                null);
+        }
+
+        return of(t);
+    }
+
+    public static <T> ContextVariable<T> untypedOf(
+        @Nullable Object value,
+        Class<T> clazz) {
+        ContextVariableType<?> type = ContextVariableTypes.getDefaultVariableTypeForClass(clazz);
+        return new ContextVariable(type, value);
+    }
+
+    public static ContextVariable<?> untypedOf(
+        @Nullable Object value,
+        ContextVariableTypeConverter<?> converter) {
+        ContextVariableType<?> type = new ContextVariableType(converter, converter.getType());
+        return new ContextVariable(type, value);
+    }
+
+    public static <T, U> ContextVariable<T> convert(
+        @Nullable
+        U it,
+        Class<T> requestedResultType) {
+        return convert(
+            it,
+            requestedResultType,
+            ContextVariableTypes.DEFAULT_TYPES
+        );
+    }
+
+    public static <T, U> ContextVariable<T> convert(
+        @Nullable
+        U it,
+        Class<T> requestedResultType,
+        ContextVariableTypes contextVariableTypes) {
+        if (it == null) {
+            return new ContextVariable<>(
+                contextVariableTypes.getVariableTypeForClass(requestedResultType),
+                null);
+        }
+
+        if (requestedResultType.isAssignableFrom(it.getClass())) {
+            try {
+                return ContextVariable.of(requestedResultType, requestedResultType.cast(it));
+            } catch (Exception e) {
+                return new ContextVariable<>(
+                    new ContextVariableType<>(
+                        new NoopConverter<>(requestedResultType),
+                        requestedResultType
+                    ),
+                    requestedResultType.cast(it)
+                );
+            }
+        }
+
+        ContextVariableType<T> requestedResultTypeVariable = null;
+
+        try {
```

Comment:
suggest breaking these out into smaller functions so you could then use streams findFirst

Reply:
Unfortunately due to the multiple exit points this is hard to extract